### PR TITLE
Add Utility Functions for URL Manipulation and Query Parameter Handling

### PR DESCRIPTION
**Pull request by andremartinsds**
This PR includes several utility functions for handling URL manipulation in Go. Each function serves a specific purpose to parse, modify, and build URLs with query parameters and other components.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new package for URL utilities, including functions for building URLs, adding query parameters, validating URLs, extracting domains, and retrieving query parameters.
- **Tests**
	- Added a comprehensive suite of unit tests to ensure the reliability and functionality of the new URL utilities, covering various scenarios and error handling.
- **Chores**
	- Updated module dependencies, including an upgrade to the `golang.org/x/text` module and the addition of a new indirect dependency on `golang.org/x/net`.

### CODE REVIEW

**Comment by andremartinsds** [comment]
Hey, :) @kashifkhan0771 can you review my PR

File: url/url.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+	}
+	if host == "" {
+		errMessage = append(errMessage, "host is required")
+	}
+
+	if path != "" {
+		re := regexp.MustCompile("^[a-zA-Z]+(\\/[a-zA-Z]+)*$")
+		if !re.MatchString(path) {
+			errMessage = append(errMessage, "path is permitted with a-z character and multiple path segments")
+		}
```
[the remark this answers is not in the document]
Comment (andremartinsds) [reply url/url.go:79]: This suggestion is not correct.

**Comment by andremartinsds** [comment]
This AI suggests changes, and once we adjust them, it requests more changes on top of what it suggested earlier. This makes what should be simple quite complicated.

**Review by kashifkhan0771 — APPROVED** [review]

**Comment by kashifkhan0771** [comment]
Thanks @andremartinsds for the contribution!

**Pull request merged**